hook.js

```diff
@@ -91,6 +91,10 @@ function isStarExportLine (line) {
   return /^\* from /.test(line)
 }
 
+function isBareSpecifier (specifier) {
+  return /^[a-zA-Z@]/.test(specifier) && !specifier.startsWith('file:')
```

Comment:
are there are other absolute specifiers we need to watch out for here? `data:` and `node:` comes to my mind here.

Also I guess if someone is using a custom loader they can put any kind of specifier right? 😬 Like the [https one](https://nodejs.org/api/module.html#import-from-https) in the Node.js docs.

Maybe it's safer to just search for `:`?

Reply:
Ah good point 🤔

Thinking about all different variants passed through here:
https://github.com/DataDog/import-in-the-middle/blob/00b01fff1f5b69dd25e307593ec54d1d8abb4844/hook.js#L157

We actually only to want to skip this code and resolve for bare specifiers. All other protocols including `file:` will be passed straight through the `URL` constructor whatever `srcUrl` and hit the next loader. The second URL constructor parameter `base` is only used if the first parameter is a relative reference and that's what trips with bare specifiers.

So we can drop the `&& !specifier.startsWith('file:')`!

Reply:
No you're correct, I need to exclude all valid URLs that match the regex!